### PR TITLE
IOS7: Implement port specific options widget

### PR DESCRIPTION
**Pull request by larsamannen**
Implement use of port specific options widget which may be used to enable/disable mouse modes.
Also implement support of Virtual controller (onscreen controls) with one configuration with a dPad and A and B buttons.

### CODE REVIEW

File: backends/platform/ios7/ios7_options.cpp [hunk 1]
```diff
@@ -91,17 +64,33 @@ void IOS7OptionsWidget::handleCommand(GUI::CommandSender *sender, uint32 cmd, ui
...
-Two finger tap: Right mouse click
-Two finger double tap: ESC
-Two finger swipe (bottom to top): Toggles Click and drag mode
-Two finger swipe (left to right): Toggles between touch direct mode and touchpad mode
-Two finger swipe (top to bottom): Global Main Menu
-Three finger swipe: Arrow keys
-Pinch gesture: Enables/disables keyboard
-Keyboard spacebar: Pause"));
+#if TARGET_OS_IOS
+			"Gestures and controls:\n"
```
Comment (sev-) [inline backends/platform/ios7/ios7_options.cpp:68]: You must wrap both strings into _(), so they are picked up by gettext

and add this file to POTFILES

File: backends/platform/ios7/ios7_osys_events.cpp [hunk 2]
```diff
@@ -500,6 +503,14 @@ bool OSystem_iOS7::handleEvent_swipe(Common::Event &event, int direction, int to
 			return false;
 		}
 
+		case kUIViewSwipeLeft: {
+			// Swipe left
+			bool enable = !ConfMan.getBool("onscreen_control");
+			ConfMan.setBool("onscreen_control", enable);
```
Comment (sev-) [inline backends/platform/ios7/ios7_osys_events.cpp:509]: You need also call ConfMan.flush() for persisting the state

File: backends/platform/ios7/ios7_osys_main.cpp [hunk 3]
```diff
@@ -300,6 +296,19 @@ void OSystem_iOS7::delayMillis(uint msecs) {
 	usleep(msecs * 1000);
 }
 
+GUI::OptionsContainerWidget *OSystem_iOS7::buildBackendOptionsWidget(GUI::GuiObject *boss, const Common::String &name, const Common::String &target) const {
+	return new IOS7OptionsWidget(boss, name, target);
+}
+
+void OSystem_iOS7::registerDefaultSettings(const Common::String &target) const {
+	ConfMan.registerDefault("onscreen_control", false);
+	ConfMan.registerDefault("touchpad_mode", true);
```
Comment (sev-) [inline backends/platform/ios7/ios7_osys_main.cpp:305]: I thought the original code had some logic around screen dimensions for setting up the default value. Was this lost?

**Review by sev- — CHANGES_REQUESTED** [review]
Added couple notes so far

File: backends/platform/ios7/ios7_osys_main.cpp (hunk 3 above)
Comment (larsamannen) [reply backends/platform/ios7/ios7_osys_main.cpp:305]: I removed that, but I can re-add it.

**Review by sev- — COMMENTED** [review]
I reviewed now more thoroughly, no further notes besides me clarifying my earlier comments

File: backends/platform/ios7/ios7_gamepad_controller.mm [hunk 4]
```diff
@@ -29,6 +29,12 @@
 
 @implementation GamepadController {
 	GCController *_controller;
+#if TARGET_OS_IOS
+	API_AVAILABLE(ios(15.0))
+	GCVirtualController *_virtualController;
+	API_AVAILABLE(ios(15.0))
+	GCVirtualControllerConfiguration *_config;
+#endif
```
Comment (criezy) [inline backends/platform/ios7/ios7_gamepad_controller.mm:37]: This does not compile with Xcode 12.4 / SDK iOS 14.4:
```
scummvm/backends/platform/ios7/ios7_gamepad_controller.mm:34:2: Unknown type name 'GCVirtualController'
scummvm/backends/platform/ios7/ios7_gamepad_controller.mm:36:2: Unknown type name 'GCVirtualControllerConfiguration'
scummvm/backends/platform/ios7/ios7_gamepad_controller.mm:54:15: Use of undeclared identifier 'GCVirtualControllerConfiguration'
scummvm/backends/platform/ios7/ios7_gamepad_controller.mm:73:27: Use of undeclared identifier 'GCVirtualController'
```

**Review by criezy — CHANGES_REQUESTED** [review]
This breaks compilation when using SDK for iOS 14 (or below).

**Comment by sev-** [comment]
Thanks, merging

**Pull request merged**